### ISSUE TITLE
Emitting directory crashes

### ISSUE THREAD

**Issue by atasak** [dsherret/ts-simple-ast#195]
When emitting a directory as described in #182 #183, a crash happens. From #182:

```typescript
const genSrc = ast.getDirectoryOrThrow("genSrc");
const genDist = genSrc.copy("../genDist");

// do manipulations on genDist's source files here

await genDist.saveUnsavedSourceFiles();

const result = await genDist.emit({
    declarationDir: "../genSrc",
    outDir: "../../dist/genDist"
});
// if you need the emitted file paths, they exist in result.getOutputFilePaths()
```

```
  Error: ENOENT: no such file or directory, open 'root/dist/genDist/tests.js.map'
    at Object.fs.openSync (fs.js:663:18)
    at Object.fs.writeFileSync (fs.js:1314:33)
    at DefaultFileSystemHost.writeFileSync (/home/niek/projects/projects/apione/node_modules/ts-simple-ast/dist/fileSystem/DefaultFileSystemHost.js:48:12)
    at Directory.emitSync (/home/niek/projects/projects/apione/node_modules/ts-simple-ast/dist/fileSystem/Directory.js:355:28)
    at ast.forgetNodesCreatedInBlock (/home/niek/projects/projects/apione/dist/compiler/compiler/schemer.js:75:37)
    at CompilerFactory.forgetNodesCreatedInBlock (/home/niek/projects/projects/apione/node_modules/ts-simple-ast/dist/factories/CompilerFactory.js:400:13)
    at TsSimpleAst.forgetNodesCreatedInBlock (/home/niek/projects/projects/apione/node_modules/ts-simple-ast/dist/TsSimpleAst.js:390:37)
    at Schemer.emit (/home/niek/projects/projects/apione/dist/compiler/compiler/schemer.js:72:18)
    at ApiOneCompiler.run (/home/niek/projects/projects/apione/dist/compiler/compiler/compiler.js:42:26)
    at Object.<anonymous> (/home/niek/projects/projects/apione/dist/compiler/runner.js:6:10)
``` 

Note: I only tested this with `genDist.saveUnsavedSourceFilesSync()` because of #194. Let me know if you can't reproduce or need more information.

**Comment by dsherret** [comment]
Thanks for reporting this! I should have tested this on an actual file system instead of a mocked one. Workaround is to manually create the `root/dist/genDist/` folder before emitting.

**Comment by atasak** [comment]
You're right, but it should also be empty. When a `root/dist/genDist/` exists but has a previous emit in it, it will also crash.

**Comment by dsherret** [comment]
This will be fixed in 5.0.0... which will be released probably tomorrow or the next day.

A temp workaround is to open up `node_modules/ts-simple-ast/dist/fileSystem/DefaultFileSystemHost.js` and add the line `FileUtils.ensureDirectoryExistsSync(this, FileUtils.getDirPath(filePath));` to both `writeFile` and `writeFileSync`
